### PR TITLE
Fixed CoreOS cluster starting guide

### PR DESCRIPTION
**Pull request by pires**
- Updated Kubernetes to version 0.8.1
- Updated CoreOS version used for AWS and GCE node provisioning
- Added a semaphore script to prevent (failed) writes to `etcd` before it's ready - which would result in `flannel` failure and consequently borked Kubernetes installation.

### CODE REVIEW

**Comment by jmreicha** [comment]
I used these (modified) configs to get a fresh cluster up and running on v0.8.1 today.

One thing that came up in the IRC channel that I'm not sure about is the `ExecStartPre=-/usr/bin/etcdctl mk /coreos.com/network/config '{"Network":"10.244.0.0/16", "Backend": {"Type": "vxlan"}}'` command.

@luqmansahaf suggested using "set" rather than "mk" to set key, both seem to work though.

**Comment by yhsiang** [comment]
I used on vultr, it works fine :+1:

**Comment by brendandburns** [comment]
This AMI image is already out of date... I used ami-e12679d1 to validate.

Should we add a note to the docs indicating that user's may need to update the AMI and pointing them at:

https://coreos.com/docs/running-coreos/cloud-providers/ec2/

?

**Comment by brendandburns** [comment]
I tried this out and it seems to work for me with the exception of updating the AMI image.

@pires any chance you can update the docs to point out the potential need to update that image?

Then I'll be happy to merge.

**Comment by pires** [comment]
@brendandburns sure thing! On my way to it.

**Comment by erictune** [comment]
It looks like https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/aws/cloudformation-template.json contains inline copies of the node.yaml and master.yaml files that you are updating in this PR.  Can that be updated by this PR too, or at least an issue filed?

**Comment by pires** [comment]
@brendandburns anything else?

**Comment by pires** [comment]
@erictune sure thing. On my way to it.

**Comment by pires** [comment]
@erictune I won't be changing the AMI image from that file, just the inline cloud-configs. Is that OK?

File: docs/getting-started-guides/coreos/coreos_multinode_cluster.md [hunk 1]
```diff
@@ -45,7 +47,7 @@ Edit `node.yaml` and replace all instances of `<master-private-ip>` with the pri
 ```
 aws ec2 run-instances \
 --count 1 \
---image-id ami-d92377e9 \
+--image-id ami-c484f5ac \
```
Comment (brendandburns) [inline docs/getting-started-guides/coreos/coreos_multinode_cluster.md]: <ami_image_id> here too.

**Comment by brendandburns** [comment]
Missed one AMI id, other than that LGTM, will merge once that's fixed.

File: docs/getting-started-guides/coreos/coreos_multinode_cluster.md (hunk 1 above)
Comment (pires) [inline docs/getting-started-guides/coreos/coreos_multinode_cluster.md]: Nice catch, thanks. Damn search/replace.

**Comment by pires** [comment]
@brendandburns done!

**Comment by brendandburns** [comment]
LGTM, merging.

**Comment by jmreicha** [comment]
@pires just a heads up, this might need updated again soon.  CoreOS 561.0.0 will ship with ext 4 so the docker daemon should use the `-s=overlay` flag on worker/minion nodes.  I was testing this weekend.

**Comment by pires** [comment]
@jmreicha be my guest and go ahead and try it. If it works, submit a PR ;-)

**Comment by erictune** [comment]
#3625 changes to overlay

**Comment by pires** [comment]
I've changed missing ```standalone.yaml``` as well in #3639.

**Pull request merged**